File: my_cpp_pkg/src/add_two_ints_client.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"


class AddTwoIntsClientNode: public rclcpp::Node
{
    public:
        AddTwoIntsClientNode(): Node("add_two_ints_client")
        {
            threads.push_back(std::thread(std::bind(&AddTwoIntsClientNode::callAddTwoIntsService, this, 1, 3)));
            threads.push_back(std::thread(std::bind(&AddTwoIntsClientNode::callAddTwoIntsService, this, 5, 3)));
            threads.push_back(std::thread(std::bind(&AddTwoIntsClientNode::callAddTwoIntsService, this, 3, 3)));
        }

        void callAddTwoIntsService(int a, int b)
        {
            auto client = this->create_client<example_interfaces::srv::AddTwoInts>("add_two_ints");
            while (!client->wait_for_service(std::chrono::seconds(1)))
            {
                RCLCPP_WARN(this->get_logger(), "Waiting for server add_two_ints");
            }

            auto request = std::make_shared<example_interfaces::srv::AddTwoInts::Request>();
            request->a = a;
            request->b = b;

            auto future = client->async_send_request(request);

            try {
                auto response = future.get();
                RCLCPP_INFO(this->get_logger(), "%d + %d = %ld", a, b, response->sum);
            } catch (const std::exception &e) {
                //std::string error = e.what();
                //RCLCPP_ERROR(this->get_logger(), "Error while calling the service %s", error.c_str());
                RCLCPP_ERROR(this->get_logger(), "Error while calling the service");
            }
        }

    private:
        std::vector<std::thread> threads;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<AddTwoIntsClientNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: my_py_pkg/my_py_pkg/add_two_ints_client.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts
from functools import partial

class AddTwoIntsClientNode(Node):
    def __init__(self):
        super().__init__("add_two_ints_client")
        self.call_add_two_ints_server(1, 2)
        self.call_add_two_ints_server(1, 5)
        self.call_add_two_ints_server(9, 2)

    def call_add_two_ints_server(self, a, b):
        client = self.create_client(AddTwoInts, "add_two_ints")
        while not client.wait_for_service(1.0):
            self.get_logger().warn("Waiting for service add_two_ints")
        request = AddTwoInts.Request()
        request.a = a
        request.b = b
        #response
        future = client.call_async(request)
        future.add_done_callback(partial(self.callback_add_two_ints, a=a, b=b))

    def callback_add_two_ints(self, future, a, b):
        try:
            response = future.result()
            self.get_logger().info("response from service add_two_ints: " + str(a) + " + " + str(b) + " = " + str(response.sum))
        except Exception as e:
            self.get_logger().error("Error getting response from service add_two_ints: %r" % (e,))  

def main(args = None):
    rclpy.init(args = args)

    node = AddTwoIntsClientNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()